src/File/MoveUploadedFile.php

```diff
@@ -0,0 +1,47 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Filter\File;
+
+use ErrorException;
+
+use function is_uploaded_file;
+use function move_uploaded_file;
+use function rename;
+use function restore_error_handler;
+use function set_error_handler;
+
+/** @internal */
+final class MoveUploadedFile implements MoveUploadedFileInterface
+{
+    public function moveUploadedFile(string $sourceFile, string $targetFile): void
+    {
+        set_error_handler(static function (
+            int $errorNumber,
+            string $message,
+            $file = null,
+            $line = null,
+        ): never {
+            throw new ErrorException($message, $errorNumber, $errorNumber, $file, $line);
+        });
+
+        if (! is_uploaded_file($sourceFile)) {
```

Comment:
Isn't this the security feature of `move_uploaded_file` that will be disabled with this condition?
This Filter should be used for uploaded files only. If someone hijack the $_FILES object this security mechanism will not work anymore. 

You need this only for the Tests and there is already the MoveUploadedFileInterface. Would prefer to use here the move_uploaded_file method and use the rename Handler only for the tests

Reply:
Good point, running `is_uploaded_file` in the filter and rejecting input upfront would cover this. We could then continue to use the existing (And internal) `MoveUploadedFile` implementation in the knowledge that the conditional there is only used in test scenarios.

Reply:
Brains aren't working well today - thanks for the review @marcelthole - I've reverted closer to your original work 👍